src/rowtransforms.jl

```diff
@@ -0,0 +1,60 @@
+struct NormalizeRow <: DataAugmentation.Transform
+	normstats
+	normcols
+end
+
+struct Categorify <: DataAugmentation.Transform
+	pooldict
+	categorycols
+end
+
+struct FillMissing <: DataAugmentation.Transform
+	fmvals
+	contcols
+	catcols
+end
+
+function DataAugmentation.apply(tfm::FillMissing, item::TabularItem; randstate=nothing)
+	x = (; zip(item.columns, [data for data in item.data])...)
+	for col in tfm.contcols
+		if ismissing(x[col])
+			Setfield.@set! x[col] = tfm.fmvals[col]
+		end
+	end
+	for col in tfm.catcols
+		if ismissing(x[col])
+			Setfield.@set! x[col] = "missing"
+		end
+	end
+	TabularItem(x, item.columns)
+end
+
+function DataAugmentation.apply(tfm::NormalizeRow, item::TabularItem; randstate=nothing)
+	x = (; zip(item.columns, [data for data in item.data])...)
+	for col in tfm.normcols
+		colmean, colstd = tfm.normstats[col]
+		Setfield.@set! x[col] = (x[col] - colmean)/colstd
+	end
+	TabularItem(x, item.columns)
+end
+
+function DataAugmentation.apply(tfm::Categorify, item::TabularItem; randstate=nothing)
+	x = (; zip(item.columns, [data for data in item.data])...)
+	for col in tfm.categorycols
```

Comment:
Same comment here about double iteration of the columns

Reply:
Did you mean something like this?
```julia
function tfmrowvals(tfm::NormalizeRow, col, val)
    if col in tfm.cols
        colmean, colstd = tfm.dict[col]
        val = (val - colmean)/colstd
    end
    (col, val)
end

function apply(tfm::NormalizeRow, item; randstate=nothing)
    TabularItem((; 
            tfmrowvals.(
                [tfm for _ in 1:length(item.columns)],
                item.columns,
                [val for val in item.data])...
            ), 
        item.columns
    )
end
```

Reply:
If this is better than the current implementation, we can even have a single `apply` which works on `Union` of all the transforms, and different methods for `tfmrowvals`.

Reply:
Probably still need to dispatch on each type separately in order to know which `tfmrowvals` function to call, right?

Reply:
Yeah there will probably be 3 `tfmrowvals` methods.

Reply:
```julia
function apply(tfm::NormalizeRow, item; randstate=nothing)
    x = NamedTuple(Iterators.map(item.cols, item.data) do col, val
        if col in tfm.cols
            colmean, colstd = tfm.dict[col]
            val = (val - colmean)/colstd
        end
        (col, val)
    end)
end
```